code/modules/mob/interactive.dm

```diff
@@ -527,11 +603,25 @@
 /mob/living/carbon/human/interactive/proc/shitcurity(obj)
 	if(retal && TARGET)
 		for(var/obj/item/I in src.contents)
-			if(istype(I,/obj/item/weapon/restraints/handcuffs))
+			if(istype(I,/obj/item/weapon/restraints))
 				take_to_slot(I)
 				I.attack(TARGET,src)
 				sleep(25)
 
+/mob/living/carbon/human/interactive/proc/dojanitor(obj)
+	favoured_types = list(/obj/item/weapon/mop, /obj/item/weapon/reagent_containers/glass/bucket, /obj/item/weapon/reagent_containers/spray/cleaner, /obj/effect/decal/cleanable)
+	if(istype(main_hand,/obj/item/weapon/mop))
+		var/obj/item/weapon/mop/M = main_hand
+		if(M)
+			if(M.reagents.total_volume <= 5)
+				M.reagents.add_reagent("water", 25) // bluespess water delivery for AI

```

Comment:
Using a bucket is one level of complication that would probably not work at all.

Having an NPC find a bucket, then find a water source, fill it and then clean is like monkey's and typewriters.

If it's really an issue then I can try, but I wouldn't put any hope in it.